sdk/resourcemanager/azure-resourcemanager-servicebus/src/main/java/com/azure/resourcemanager/servicebus/implementation/TimeSpan.java

```diff
@@ -0,0 +1,450 @@
+
+/**
+ * Represents a time interval.
+ */
+public class TimeSpan {
+    private static long ticksPerMillisecond = 10000;
+    private static long ticksPerSecond = ticksPerMillisecond * 1000;
+    private static long ticksPerMinute = ticksPerSecond * 60;
+    private static long ticksPerHour = ticksPerMinute * 60;
+    private static long ticksPerDay = ticksPerHour * 24;
+    private static double millisecondsPerTick = 1.0 / ticksPerMillisecond;
+    private static double secondsPerTick = 1.0 / ticksPerSecond;
+    private static double minutesPerTick = 1.0 / ticksPerMinute;
+
+    private long days;
+    private long hours;
+    private long minutes;
+    private long seconds;
+    private long milliseconds;
+
+    /**
+     * Creates TimeSpan.
+     */
+    public TimeSpan() {
+    }
+
+    /**
+     * Specifies the number of days.
+     *
+     * @param days number of days
+     * @return TimeSpan
+     */
+    public TimeSpan withDays(final long days) {
+        this.days = days;
+        return this;
+    }
+
+    /**
+     * Specifies the number of hours.
+     *
+     * @param hours number of hours
+     * @return TimeSpan
+     */
+    public TimeSpan withHours(final long hours) {
+        this.hours = hours;
+        return this;
+    }
+
+    /**
+     * Specifies the number of minutes.
+     *
+     * @param minutes number of minutes
+     * @return TimeSpan
+     */
+    public TimeSpan withMinutes(final long minutes) {
+        this.minutes = minutes;
+        return this;
+    }
+
+    /**
+     * Specifies the number of seconds.
+     *
+     * @param seconds number of seconds
+     * @return TimeSpan
+     */
+    public TimeSpan withSeconds(final long seconds) {
+        this.seconds = seconds;
+        return this;
+    }
+
+    /**
+     * Specifies the number of milliseconds.
+     *
+     * @param milliseconds number of milliseconds
+     * @return TimeSpan
+     */
+    public TimeSpan withMilliseconds(final long milliseconds) {
+        this.milliseconds = milliseconds;
+        return this;
+    }
+
+    /**
+     * @return days value
+     */
+    public long days() {
+        return this.days;
+    }
+
+    /**
+     * @return hours value
+     */
+    public long hours() {
+        return this.hours;
+    }
+
+    /**
+     * @return minutes value
+     */
+    public long minutes() {
+        return this.minutes;
+    }
+
+    /**
+     * @return seconds value
+     */
+    public long seconds() {
+        return this.seconds;
+    }
+
+    /**
+     * @return mill-seconds value
+     */
+    public long milliseconds() {
+        return this.milliseconds;
+    }
+
+    /**
+     * @return total number of milliseconds represented by this instance
+     */
+    public double totalMilliseconds() {
+        return totalTicks() * millisecondsPerTick;
+    }
+
+    /**
+     * @return total number of milliseconds represented by this instance in duration format
+     */
+    public Duration totalMillisDuration() {
```

Comment:
Duration is just Duration. It is about time, not about unit. Maybe just `toDuration()`?